tests/phpunit/tests/test-rest-languages.php

```diff
@@ -0,0 +1,243 @@
+
+		$this->assertArrayHasKey( 'id', $data );
+		$this->assertSame( $language->term_id, $data['id'] );
+		$this->assertArrayHasKey( 'order', $data );
+		$this->assertSame( $language->term_group, $data['order'] );
+		$this->assertArrayHasKey( 'flag_tag', $data );
+		$this->assertSame( $language->flag, $data['flag_tag'] );
+		$this->assertArrayHasKey( 'custom_flag_tag', $data );
+		$this->assertSame( $language->custom_flag, $data['custom_flag_tag'] );
+		$this->assertArrayHasKey( 'is_active', $data );
+		$this->assertSame( $language->active, $data['is_active'] );
+
+		// Single check to make sure other fields are not missing.
+		$this->assertArrayHasKey( 'term_props', $data );
+		$this->assertSame( $language->get_tax_props(), $data['term_props'] );
+
+		// Check the default category.
+		$def_cat_lang = $this->pll_env->model->term->get_language( $def_cat_id );
+		$this->assertInstanceOf( PLL_Language::class, $def_cat_lang );
+		$this->assertSame( $data['code'], $def_cat_lang->slug );
+
+		// 2- Create a language with custom values.
+		$values = array(
+			'name'      => 'François',
+			'code'      => 'fra',
+			'direction' => 'rtl',
+			'flag'      => 'be',
+			'order'     => 22,
+		);
+		$request = new WP_REST_Request( 'POST', '/pll/v2/languages' );
+		$request->set_param( 'locale', 'fr_FR' ); // Required.
+		$request->set_param( 'w3c', 'foo' ); // Should not be set.
+		$request->set_param( 'set_default_cat', false ); // Do not create the default category for this language.
+
+		foreach ( $values as $name => $value ) {
+			$request->set_param( $name, $value );
+		}
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+
+		// Check the values match.
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+
+		foreach ( $values as $name => $value ) {
+			$this->assertSame( $value, $data[ $name ] );
+		}
+
+		$this->assertNotSame( 'foo', $data['w3c'] );
+
+		// Check the default category is not created.
+		$def_cat_id_fr = $this->pll_env->model->term->get( $def_cat_id, $data['code'] );
+		$this->assertSame( 0, $def_cat_id_fr );
+	}
+
+	public function test_get_language() {
+		self::create_language( 'fr_FR' );
+		self::create_language( 'en_US' );
+
+		$request = new WP_REST_Request( 'GET', '/pll/v2/languages/fr' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertIsArray( $data );
+		$this->assertArrayHasKey( 'code', $data );
+		$this->assertSame( 'fr', $data['code'] );
+	}
+
+	public function test_update_language() {
+		wp_set_current_user( self::$administrator );
+		self::create_language( 'fr_FR' );
+
+		$values = array(
+			'locale'    => 'fr_BE',
+			'name'      => 'François',
+			'new_code'  => 'fra',
+			'direction' => 'rtl',
+			'flag'      => 'be',
+			'order'     => 22,
+		);
+		$request = new WP_REST_Request( 'PATCH', '/pll/v2/languages/fr' );
```

Comment:
We should also test with `PUT` IMHO.
One of these accept only some fields.

Reply:
Done in [01cc7b0](https://github.com/polylang/polylang/pull/1505/commits/01cc7b08079755bc0d0840c47fa1e005c34a863f).